lib/ae_mdw/names.ex

```diff
@@ -549,6 +553,43 @@ defmodule AeMdw.Names do
     end
   end
 
+  defp paginate_nested_resource(bi_txis, scope, cursor, pagination) do
+    cursor = deserialize_nested_cursor(cursor)
+
+    bi_txis =
+      case scope do
+        {:gen, first_gen..last_gen} ->
+          bi_txis
+          |> Enum.drop_while(fn {{height, _mbi}, _txi} -> height >= last_gen end)
+          |> Enum.take_while(fn {{height, _mbi}, _txi} -> height <= first_gen end)
+
+        nil ->
+          bi_txis
+      end
+
+    streamer = fn
+      :forward ->
+        bi_txis = Enum.reverse(bi_txis)
+
+        if cursor do
+          Enum.drop_while(bi_txis, &(&1 < cursor))
+        else
+          bi_txis
+        end
+
+      :backward ->
+        if cursor do
+          Enum.drop_while(bi_txis, &(&1 > cursor))
+        else
+          bi_txis
+        end
+    end
```

Comment:
candidate for a function like `build_streamer`

Reply:
Didn't think it was necessary, since it wouldn't have different clauses and it would be used only once. This avoids having to jump to different places to understand what a function does, so I would rather leave it as it is